### PR TITLE
first commit of tm4c123bsp, validate on MDK5

### PR DESCRIPTION
**Pull request by LYH-ux**
## 拉取/合并请求描述：(PR description)

[
1.提交原因：现有bsp中未支持TM4C123，故请求提求本tm4c123bsp
2.解决的问题：在bsp文件夹下新增tm4c123bsp，本bsp是对TI   EK-tm4c123GXL开发板的支持。
3.解决方案：基于stm32下的bsp制作教程，制作本bsp，基本符合bsp规范，无多余文件。
                     支持env 和 sconscript构建脚本；
                     支持设备驱动：PIN  UART  PWM  SPI  ADC
4.在Keil MDK5环境和ICDI仿真条件下进行了测试，可使用env配置和scons构建工程，编译后可直接使用。
已测试驱动： 

- [x] PIN  

- [x] UART  

- [x] PWM 

- [x]  SPI  

- [x] ADC
设备驱动为基础版本， 未支持DMA。
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [x] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [ ] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other style
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [ ] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality

### CODE REVIEW

**Comment by LYH-ux** [comment]
pull request for tm4c123bsp

File: bsp/tm4c123bsp/board/board.c [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+// frequency supplied to the SysTick timer and the processor 
+// core clock.
+uint32_t SystemCoreClock;
+
+
+
+void SystemCoreClockUpdate(void)
+{
+	
+	   FPULazyStackingEnable();
```
Comment (BernardXiong) [inline bsp/tm4c123bsp/board/board.c:37]: please take care of code style, such as indentation.

File: bsp/tm4c123bsp/board/board.c [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    /* enter interrupt */
+    rt_interrupt_enter();
+
+    rt_tick_increase();
+
+    /* leave interrupt */
+    rt_interrupt_leave();
+}
+
+
```
Comment (BernardXiong) [inline bsp/tm4c123bsp/board/board.c:104]: No two empty line.

File: bsp/tm4c123bsp/board/tm4c123_config.c [hunk 3]
```diff
@@ -0,0 +1,77 @@
+#include "stdbool.h"
```
Comment (BernardXiong) [inline bsp/tm4c123bsp/board/tm4c123_config.c:1]: What's the copyright and changelog information in file header?

File: bsp/tm4c123bsp/libraries/Drivers/drv_flash/drv_flash_f1.c [hunk 4]
```diff
@@ -0,0 +1,197 @@
+/*
```
Comment (BernardXiong) [inline bsp/tm4c123bsp/libraries/Drivers/drv_flash/drv_flash_f1.c:1]: Is F1 series also available in TI MCU?

File: bsp/tm4c123bsp/libraries/Drivers/drv_uart.c [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+	
+#ifdef BSP_USING_UART3
+    UART3_CONFIG,
+#endif	
+};
+
+static struct tm4c123_uart uart_obj[sizeof(uart_config) / sizeof(uart_config[0])] = {0};
+
+
+//struct serial_configure
```
Comment (BernardXiong) [inline bsp/tm4c123bsp/libraries/Drivers/drv_uart.c:86]: If these codes are unnecessary, remove them.

File: bsp/tm4c123bsp/libraries/Drivers/config/adc_config.h [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+#endif
+
+#ifdef BSP_USING_ADC0
+#ifndef ADC0_CONFIG
+#define ADC0_CONFIG                             \
+    {                                           \
+        .name                    ="adc0",       \
+        .adcbase                 = ADC0_BASE,    \
+        .channel                 = 0 ,           \
+		.sequence                = 2 ,           \
```
Comment (BernardXiong) [inline bsp/tm4c123bsp/libraries/Drivers/config/adc_config.h:25]: indentation issue...

File: bsp/tm4c123bsp/board/board.c (hunk 1 above)
Comment (LYH-ux) [reply bsp/tm4c123bsp/board/board.c]: Thank you . I'v  format the code to obtain the code style.

File: bsp/tm4c123bsp/board/board.c (hunk 2 above)
Comment (LYH-ux) [reply bsp/tm4c123bsp/board/board.c]: Thank you for your detailed suggestions.  I got it.

File: bsp/tm4c123bsp/libraries/Drivers/drv_flash/drv_flash_f1.c (hunk 4 above)
Comment (LYH-ux) [reply bsp/tm4c123bsp/libraries/Drivers/drv_flash/drv_flash_f1.c]: Sorry,I ignored it.  Now I have removed it.

File: bsp/tm4c123bsp/libraries/Drivers/drv_uart.c (hunk 5 above)
Comment (LYH-ux) [reply bsp/tm4c123bsp/libraries/Drivers/drv_uart.c]: OK. I've modified  the entire code to check and correct these issues .

**Pull request merged**